Scripts/ShaderReloadSoakTest.bv.lua

```diff
@@ -0,0 +1,38 @@
+----------------------------------------------------------------------------------------------------
+--
+-- Copyright (c) Contributors to the Open 3D Engine Project.
+-- For complete copyright and license terms please see the LICENSE at the root of this distribution.
+--
+-- SPDX-License-Identifier: Apache-2.0 OR MIT
+--
+--
+--
+----------------------------------------------------------------------------------------------------
+
+-- WARNING: This is a soak test, do not add this test to the fully automated
+-- test suites.
+
+OpenSample('RPI/ShaderReloadSoakTest')
+ResizeViewport(500, 500)
+IdleSeconds(2)
+
+function TestButton(buttonName, shaderReloadWaitSeconds, pixelCaptureWaitSeconds)
+    SetImguiValue(buttonName, true)
+    IdleSeconds(shaderReloadWaitSeconds) -- Idle for a bit to give time for the shader to reload
+    SetShowImGui(false)
+    SetImguiValue('Check color', true)
+    IdleSeconds(pixelCaptureWaitSeconds)
+    SetShowImGui(true)
+end
+
+local shaderReloadWaitSeconds = 3
+local pixelCaptureWaitSeconds = 1
```

Comment:
Did you try using the AssetTracking_* functions like MaterialHotReloadTest? Having arbitrary timeouts makes things take longer than necessary, and it doesn't scale well across platforms where shader compilation can take much longer (like on Mac IIRC).

Reply:
Will take a look.